controllers/storagecluster/storagecluster_controller.go

```diff
@@ -144,28 +148,91 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	noobaaIgnoreTimeUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*nbv1.NooBaa)
+			newObj := e.ObjectNew.(*nbv1.NooBaa)
+
+			for _, cond := range oldObj.Status.Conditions {
+				cond.LastHeartbeatTime = metav1.Time{}
+				cond.LastTransitionTime = metav1.Time{}
+			}
+
+			for _, cond := range oldObj.Status.Conditions {
```

Comment:
instead of using a `for loop` we do have a better and faster comparison using resourceQtycompare,

cmpopts is package provide by go here can use it https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#IgnoreFields

```
resourceQtyComparer = cmpopts.IgnoreFields(corev1.NodeCondition{}, "LastHeartbeatTime")
	diff1 := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
```

For more details refer to https://github.com/rook/rook/pull/14186, we have just included the similar checks for the node reconcile

Reply:
Done